src/GridFS/StreamWrapper.php

```diff
@@ -53,10 +53,11 @@ class StreamWrapper
 
     public function __destruct()
     {
-        /* This destructor is a workaround for PHP trying to use the stream well
-         * after all objects have been destructed. This can cause autoloading
-         * issues and possibly segmentation faults during PHP shutdown. */
-        $this->stream = null;
```

Comment:
I could remove this line introduced by PHPLIB-345 because even if PHP tries to close the stream again, this is no-op.

Reply:
Note that the new test reproduces also the issue described in PHPLIB-345.

Reply:
I'd have to go back and check, but PHP may also have fixed this issue by shutting down streams earlier in the shutdown process. Either way, manually closing the stream definitely fixes the write issue, but if we want to be absolutely sure we can also reset the `stream` property as was done previously to ensure the issue does not re-appear.

Reply:
I was able to reproduce the "autoloading" issue by removing the `__destruct()` function. So I don't see what would have changed in PHP core for this.

Reply:
Ah, I see. Since you've confirmed that the current code works around the issue, no objections to merging this as is 👍